File: back-end/src/main/java/es/ull/project/application/usecase/container/CreateContainerUseCase.java
package es.ull.project.application.usecase.container;

import es.ull.project.domain.entity.Container;
import es.ull.project.domain.enumerate.ServiceZone;
import es.ull.project.domain.enumerate.WasteType;
import es.ull.project.domain.valueobject.capacity.ContainerCapacityLiters;
import es.ull.project.domain.valueobject.demand.DailyWasteDemandLitersPerDay;
import es.ull.project.domain.valueobject.location.Location;
import es.ull.project.domain.valueobject.name.Name;

/**
 * Use case for creating a new container.
 */
public interface CreateContainerUseCase {
    /**
     * Creates a new container with the specified attributes.
     *
     * @param name        the container name
     * @param location    the location of the container
     * @param wasteType   the type of waste the container handles
     * @param capacityLiters the container capacity in liters
     * @param dailyDemandLitersPerDay the daily waste demand in liters per day
     * @param serviceZone the service zone of the container
     * @return the created container
     */
    Container create(Name name, Location location, WasteType wasteType, ContainerCapacityLiters capacityLiters, DailyWasteDemandLitersPerDay dailyDemandLitersPerDay, ServiceZone serviceZone);
}

File: front-end/src/application/usecase/ContainerManagement/CreateContainerUseCase.ts
// CreateContainerUseCase.ts
// Use case contract for creating a new container

import type { Either, DataError } from '@ull-tfg/ull-tfg-typescript';
import { Container } from '../../../domain/entity/Container';
import { Location } from '../../../domain/valueobject/location/Location';
import { WasteType } from '../../../domain/enumerate/WasteType';
import { WasteDemand } from '../../../domain/valueobject/demand/WasteDemand';

/**
 * Use case for registering a new container in the system.
 * Input: location, waste type, waste demand, service zone
 * Output: Container (entity)
 */

// Command object for input data
export interface CreateContainerCommand {
    location: Location;
    wasteType: WasteType;
    wasteDemand: WasteDemand;
    serviceZone?: string | null;
}

// Result type for the use case
export type CreateContainerResult = Container;

// Use case contract
export interface CreateContainerUseCase {
    /**
     * Handles the creation of a new container
     *
     * @param command Data needed to create the container
     * @returns Either a DataError or the created Container
     */
    execute(command: CreateContainerCommand): Promise<Either<DataError, CreateContainerResult>>;
}
